projects/igniteui-angular/src/lib/radio/themes/_base.scss

```diff
@@ -0,0 +1,431 @@
+        background: var-get($_theme, 'fill-color');
+    }
+    
+    %radio-ripple--hover-invalid {
+        background: var-get($_theme, 'error-color');
+    }
+    
+    %radio-ripple--focused {
+        background: var-get($_theme, 'empty-color');
+        transition: background .2s $ease-out-quad;
+        opacity: .12;
+    
+        @container style(--ig-theme-variant: dark) {
+            opacity: .24;
+        }
+    }
+
+    %radio-ripple--pressed {
+        opacity: .12;
+    
+        @container style(--ig-theme-variant: dark) {
+            opacity: .24;
+        }
+    }
+
+    @include b(igx-radio) {
+        position: relative;
+        display: inline-flex;
+        flex-flow: row nowrap;
+        align-items: center;
+        color: var-get($_theme, 'label-color');
+        width: max-content;
+        cursor: pointer;
+
+        &:hover {
+            @include e(composite) {
+                &::before {
+                    background: var-get($_theme, 'hover-color');
+                }
+            }
+
+            @include e(ripple) {
+                @extend %radio-ripple--hover !optional;
+            }
+        }
+
+        &:active {
+            @include e(composite) {
+                &::before {
+                    background: var-get($_theme, 'hover-color');
+                }
+            }
+
+            @include e(ripple) {
+                @extend %radio-ripple--hover !optional;
+                @extend %radio-ripple--pressed !optional;
+            }
+        }
+        
+        @include e(input) {
+            @include hide-default();
+        }
+        
+        @include e(composite) {
+            position: relative;
+            display: inline-block;
+            width: var(--size);
+            height: var(--size);
+            min-width: var(--size);
+            line-height: var(--size);
+            color: var-get($_theme, 'label-color');
+            user-select: none;
+
+            //ripple color
+            --color: #{var-get($_theme, 'empty-color')};
+
+            &::before,
+            &::after {
+                position: absolute;
+                content: '';
+                width: var(--size);
+                height: var(--size);
+                inset-inline-start: 0;
+                top: 0;
+                border-radius: border-radius(50%);
+            }
+
+            &::before {
+                backface-visibility: hidden;
+                z-index: 1;
+                transform: scale(0);
+                transition: all .2s ease-in;
+            }
+
+            &::after {
+                border: var(--border-width) solid var-get($_theme, 'empty-color');
+                background: var-get($_theme, 'empty-fill-color');
+            }
+        }
+        
+        @include e(label) {
+            @extend %radio-label !optional;
+
+            margin-inline-start: var(--label-margin);
+
+            &:empty {
+                margin: 0;
+            }
```

Comment:
If the `%radio-label` placeholder sets `display: none` when the label container is empty, is this line really needed?